composer/utils/object_store/uc_object_store.py

```diff
@@ -212,7 +212,7 @@ def get_object_size(self, object_name: str) -> int:
             # does not work. Once fixed, we will call the files API endpoint. We currently only use this
             # function in Composer and LLM-foundry to check the UC object's existence.
             self.client.api_client.do(method='HEAD',
-                                      path=f'{self._UC_VOLUME_FILES_API_ENDPOINT}/{self.prefix}/{object_name}',
+                                      path=f'{self._UC_VOLUME_FILES_API_ENDPOINT}/{self._get_object_path(object_name)}',
```

Comment:
Can we use os.path.join(..) to avoid bugs with leading and lagging slashes?

Reply:
we can't because `os.path.join` because it will ignore previous parts of the input if one of them is an absolute path. `self._get_object_path` returns an absolute path

Spec of `os.path.join`
```
Join two or more pathname components, inserting '/' as needed. If any component is an absolute path, all previous path components will be discarded. An empty last part will result in a path that ends with a separator.
```

Fails in the run if I try `os.path.join`
```
llmfoundry.data.finetuning.dataloader._download_remote_hf_dataset('dbfs:/Volumes/finetuning/ift/train_vol', 'train')
Traceback (most recent call last):
  File "/composer/composer/utils/object_store/uc_object_store.py", line 214, in get_object_size
    self.client.api_client.do(method='HEAD',
  File "/usr/lib/python3/dist-packages/databricks/sdk/core.py", line 127, in do
    return retryable(self._perform)(method,
  File "/usr/lib/python3/dist-packages/databricks/sdk/retries.py", line 50, in wrapper
    raise err
  File "/usr/lib/python3/dist-packages/databricks/sdk/retries.py", line 29, in wrapper
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/databricks/sdk/core.py", line 230, in _perform
    raise self._make_nicer_error(response=response, message=message) from None
databricks.sdk.errors.mapping.NotFound: Not Found
```

We could use pathlib and deal with Path objects, but going to push the fstring fix for now

Reply:
Ah, that's a pain. Thanks for checking! The current code works in that case.